### ISSUE TITLE
r_bin_create should not require a fake RBinFile/RBinObject

### ISSUE THREAD

**Issue by ret2libc** [radare/radare2#12216]
When you use `rabin2 -C` to create a binary, the code uses `r_bin_use_arch` to select the binfile with the given arch/bits/plugin. However, if we are creating a file we do not really care about existing binfiles. The current code creates a fake binfile to pass to the plugins `create` method the selected arch/bits/plugin/etc.
```C
	RBinFile *binfile = r_bin_file_find_by_arch_bits (bin, arch, bits);
	RBinObject *obj = NULL;
	if (binfile) {
		....
	} else {
 		void *plugin = r_bin_get_binplugin_by_name (bin, name);
		if (plugin) {
			...
			binfile = r_bin_file_new (bin, "-", NULL, 0, 0, 0, 999, NULL, NULL, false);
			...
			obj = r_bin_object_new (binfile, plugin, 0, 0, 0, 1024);
			...
		}
	}

```

This is hacky because we create a RBinFile/RBinObject that does not respect what you would expect from a real RBinFile (e.g. it does not have any backing `bytes` or anything else). The only reason the fake RBinFile/RBinObject is created is because some plugins (e.g. bin_mach0.c) needs to know the architecture/bits/etc. to generate the right binary file.

We should fix this by introducing a RBinTarget structure which includes `pluginname`, `arch`, `bits` (and other things that may be necessary in the future) and that is passed to `r_bin_create` and down to the `create` methods of the RBinPlugins. This structure could also be used in other places where we instead use `arch`, `bits`, etc...

**Comment by radare** [comment]
your solution is also hacky because its adding more arguments to afunction that will probably need even more parameters so it will have to chagne several more times in the future, thats why  ip refer to pack all that stuff in a struct that we can extend

**Comment by ret2libc** [comment]
I absolutely agree, no need to bring more reasons ;) I also already opened an issue to track this

**Comment by ret2libc** [comment]
Ops sorry, I thought it was on the other issue... :) Why do you say it's hacky? Did you read the issue? I was suggesting, indeed, to use a structure that contains all those things

Quoting: `"We should fix this by introducing a RBinTarget structure which includes pluginname, arch, bits (and other things that may be necessary in the future) and that is passed to r_bin_create and down to the create methods of the RBinPlugins"`

**Comment by ret2libc** [comment]
This has been fixed in 300844fa51073c1aa59df7f2b37896141bfc3ca8
